src/main/kotlin/org/rust/ide/inspections/import/AutoImportFix.kt

```diff
@@ -106,23 +104,27 @@ class AutoImportFix(element: RsElement) : LocalQuickFixOnPsiElement(element), Hi
             if (path.ancestorStrict<RsUseSpeck>() != null) {
                 // Don't try to import path in use item
                 Testmarks.pathInUseItem.hit()
-                return Context(basePath, emptyList())
+                return Context(path, emptyList())
             }
 
-            val isNameInScope = path.hasInScope(basePath.referenceName, TYPES_N_VALUES)
+            val isNameInScope = path.hasInScope(path.referenceName, TYPES_N_VALUES)
             if (isNameInScope) {
                 // Don't import names that are already in scope but cannot be resolved
                 // because namespace of psi element prevents correct name resolution.
                 // It's possible for incorrect or incomplete code like "let map = HashMap"
                 Testmarks.nameInScope.hit()
-                return Context(basePath, emptyList())
+                return Context(path, emptyList())
             }
 
-            val candidates = getImportCandidates(ImportContext.from(project, path, false), basePath.referenceName, path.text) {
-                path != basePath || !(it.item is RsMod || it.item is RsModDeclItem || it.item.parent is RsMembers)
+            val candidates = getImportCandidates(
+                ImportContext.from(project, path, false),
+                path.referenceName,
+                path.text
+            ) {
+                true
```

Comment:
Not super sure about this one. From what I gathered it returns correct candidates but don't know if this is proper way to do it

Reply:
I think you don't need to change anything here